src/test/groovy/org/prebid/server/functional/tests/SchainSpec.groovy

```diff
@@ -154,4 +156,40 @@ class SchainSpec extends BaseSpec {
         def bidderRequest = bidder.getBidderRequest(bidRequest.id)
         assert bidderRequest.source?.schain?.nodes == [GLOBAL_SUPPLY_SCHAIN_NODE]
     }
+
+    def "PBS should use source.ext.schain when ext.prebid.schains.bidder doesn't requested"() {
+        given: "Pbs config"
+        def defaultPbsService = pbsServiceFactory.getService([("adapters.generic.ortb-version"): "2.5"])
+
+        and: "Default basic BidRequest with schain obj"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            source = new Source(ext: new SourceExt(schain: SupplyChain.defaultSupplyChain))
+            ext.prebid.schains = [new PrebidSchain(bidders: [UNKNOWN.value], schain: SupplyChain.defaultSupplyChain)]
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain requested source.ext.schain"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.source.ext.schain == bidRequest.source.ext.schain
+    }
+
+    def "PBS should use ext.prebid.schains.schain instead of source.ext.chain when ext.prebid.schains.bidder requested"() {
+        given: "Pbs config"
+        def defaultPbsService = pbsServiceFactory.getService([("adapters.generic.ortb-version"): "2.5"])
+
+        and: "Default basic BidRequest with schain obj"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            source = new Source(ext: new SourceExt(schain: SupplyChain.defaultSupplyChain))
+            ext.prebid.schains = [new PrebidSchain(bidders: [GENERIC.value], schain: SupplyChain.defaultSupplyChain)]
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain requested ext.prebid.schains[*].schain"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.source.ext.schain == bidRequest.ext.prebid.schains[0].schain
```

Comment:
Please extract `bidRequest.ext.prebid.schains[0].schain` to variable so that you don't do a comparison with the bid request object (at least the part of it).

Reply:
Also please make sure that the size of the array is only 1.